src/Alpha_complex/include/gudhi/Alpha_complex.h

```diff
@@ -366,14 +369,18 @@ class Alpha_complex {
    * (will be set to `NaN`).
    * Default value is `false` (which means compute the filtration values).
    *
-   * @return true if creation succeeds, false otherwise.
-   * 
+   * @return `true` if creation succeeds, `false` otherwise.
+   *
    * @pre Delaunay triangulation must be already constructed with dimension strictly greater than 0.
    * @pre The simplicial complex must be empty (no vertices)
-   * 
+   *
    * Initialization can be launched once.
+   *
+   * \note Weighted Alpha complex can have negative filtration values. If `output_squared_values` is set to `false`,
+   * filtration values will be `NaN` in this case.
```

Comment:
If I understood correctly, contrary to the regular Alpha filtration, the filtration values in a weighted Alpha filtration are not distances and the option of having $\alpha$ squared or not has not much meaning for it. It would make more sense in that case to ignore `output_squared_values`. If the user did not want the filtration values (what the NaN represent, I guess?), he would just use a simplex tree with `store_filtration` at `false`, no?

Or are the NaN needed for some cases? Or did I simply misunderstood something?

Reply:
I did so on [44a6747](https://github.com/GUDHI/gudhi-devel/pull/1172/commits/44a6747c1c2ffd0d80db14bfd65d64cd9f0e6e88), but I had to print a warning to the user in case he sets weights but don't want square filtration values.